pkgs/by-name/rq/rquickshare/package.nix

```diff
@@ -0,0 +1,22 @@
+{ appimageTools, fetchurl, lib }:
+
+let
+  version = "0.7.1";
+in
+appimageTools.wrapType2 {
+  name = "rquickshare";
+  src = fetchurl {
+    url = "https://github.com/Martichou/rquickshare/releases/download/v${version}/r-quick-share_${version}_amd64.AppImage";
+    hash = "sha256-716d7T4nbs/dDS4KVGTADCpLO31U8iq6hDVD+c7Ks1I=";
+  };
+  # extraPkgs = pkgs: with pkgs; [ ]; # no extra packages required
+  meta = with lib; {
+    description = "A Quickshare client in Rust";
+    homepage = "https://github.com/Martichou/rquickshare";
+    changelog = "https://github.com/Martichou/rquickshare/blob/master/CHANGELOG.md#071-2024-05-01"; # edit anker with each release
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ hannesgith ];
+    mainProgram = "rquickshare";
+    platforms = platforms.linux;
```

Comment:
```suggestion
    platforms = [ "x86_64-linux" ];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

Reply:
thanks for your review! 

any reason the hardcoded string is preferrred?

Reply:
The hardcoded string is not preferred, but you are using a x86_64(amd64) binary which apparently won't work on other architectures. And

```
nix-repl> lib.platforms.linux
[
  "aarch64-linux"
  "armv5tel-linux"
  "armv6l-linux"
  "armv7a-linux"
  "armv7l-linux"
  "i686-linux"
  "loongarch64-linux"
  "m68k-linux"
  "microblaze-linux"
  "microblazeel-linux"
  "mips-linux"
  "mips64-linux"
  "mips64el-linux"
  "mipsel-linux"
  "powerpc64-linux"
  "powerpc64le-linux"
  "riscv32-linux"
  "riscv64-linux"
  "s390-linux"
  "s390x-linux"
  "x86_64-linux"
]
```

Reply:
oh okay that makes a lot of sense, thanks!